File: packages/engine/src/cpus/6502/instructions/tya.js
import { CycleType } from '..'

/**
 * Transfer Y to A
 *
 * A = Y
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       TYA           $98  1   2
 */
export function* tya() {
  this.acc = this.y
  this.zero = this.acc === 0
  this.negative = (this.acc & 0x80) > 0
  yield CycleType.GET
}

File: packages/engine/src/cores/famibob/cpu/instructions/tya.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// Transfer Y to A
//
// A = Y
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       TYA           $98  1   2

pub fn tya(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.acc = self.y;
    self.z = self.acc == 0;
    self.n = (self.acc & 0x80) > 0;
}
